cli/src/commands/operation/log.rs

```diff
@@ -190,18 +196,29 @@ fn do_op_log(
     let mut formatter = ui.stdout_formatter();
     let formatter = formatter.as_mut();
     let limit = args.limit.unwrap_or(usize::MAX);
-    let iter = op_walk::walk_ancestors(slice::from_ref(current_op)).take(limit);
+    let iter = op_walk::walk_ancestors(slice::from_ref(current_op));
+
     if !args.no_graph {
         let mut raw_output = formatter.raw()?;
         let mut graph = get_graphlog(graph_style, raw_output.as_mut());
-        for op in iter {
+        let iter = iter.map(|op| -> Result<_, OpStoreError> {
             let op = op?;
-            let mut edges = vec![];
-            for id in op.parent_ids() {
-                edges.push(Edge::Direct(id.clone()));
-            }
+            let edges = op.parents().map_ok(GraphEdge::direct).try_collect()?;
+            Ok((op, edges))
+        });
+        let iter_nodes: Box<dyn Iterator<Item = _>> = if args.reversed {
+            Box::new(reverse_graph(iter)?.into_iter().map(Ok))
+        } else {
+            Box::new(iter)
+        };
+        for node in iter_nodes.take(limit) {
+            let (op, edges) = node?;
+            let graphlog_edges = edges
+                .into_iter()
+                .map(|e| convert_edge_type(e).map(|e| e.id().clone()))
```

Comment:
nit: If you're interested, can you take a look at https://github.com/jj-vcs/jj/pull/5170#discussion_r1898770980 ?

Reply:
So in this case, I should just remove the `jj_cli::graphlog::Edge` type and manually map from `jj_lib::graph::GraphEdge` to `renderdag::Ancestor` for rendering?

Reply:
I think all users of `graphlog::Edge` type can be migrated to `GraphEdge`. iirc, it existed because we had our own graph renderer based on `graphlog::Edge`.

Reply:
@yuja I've updated the PR to remove `graphlog::Edge`. Could you see if there's anything I missed?